### PR TITLE
Update to TM 0.31

### PR DESCRIPTION
**Pull request by tomtau**
noticed that 0.3* changed some ABCI defs, but they are not breaking for most requests-responses, I guess

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "abci"
-version = "0.4.3"
+version = "0.4.4"
```
Comment (Liamsi) [inline Cargo.toml]: There are a few renamings which can be considered breaking. E.g. `BlockSizeParams` -> `BlockParams`.
Hence, a minor bump would make sense here IMHO
```suggestion
version = "0.5.0"
```

File: protobuf/abci.proto [hunk 2]
```diff
@@ -207,13 +207,13 @@ message ResponseCommit {
...
-  BlockSizeParams block_size = 1;
+  BlockParams block = 1;
   EvidenceParams evidence = 2;
   ValidatorParams validator = 3;
 }
 
-// BlockSize contains limits on the block size.
-message BlockSizeParams {
+// BlockParams contains limits on the block size and timestamp.
+message BlockParams {
```
Comment (Liamsi) [inline protobuf/abci.proto]: The comment mentions the timestamp but the BlockParams msg doesn't contain anything about the timestamp.

Comment (tomtau) [reply protobuf/abci.proto]: this file is from https://github.com/tendermint/tendermint/blob/master/abci/types/types.proto#L215 changed with https://github.com/tendermint/rust-abci/blob/master/Makefile#L14

Comment (Liamsi) [reply protobuf/abci.proto]: Makes sense. Thanks! We probably forgot to update the comment there...

**Pull request merged**